archive/common/src/main/scala/uk/ac/wellcome/platform/archive/common/progress/flows/ProgressUpdateFlow.scala

```diff
@@ -0,0 +1,21 @@
+package uk.ac.wellcome.platform.archive.common.progress.flows
+
+import akka.NotUsed
+import akka.stream.scaladsl.Flow
+import uk.ac.wellcome.platform.archive.common.progress.models.{FailedProgressUpdate, ProgressUpdate}
+import uk.ac.wellcome.platform.archive.common.progress.monitor.ProgressMonitor
+
+import scala.util.Try
+
+object ProgressUpdateFlow {
+  def apply(progressMonitor: ProgressMonitor):
+  Flow[ProgressUpdate, Either[FailedProgressUpdate, ProgressUpdate], NotUsed] = {
+
+    Flow[ProgressUpdate].map(progressUpdate => {
+      Try(progressMonitor.update(progressUpdate)).toEither
+        .left.map(e => FailedProgressUpdate(e, progressUpdate))
+        .right.map(_ => progressUpdate)
+    })
```

Comment:
I wonder if the progressMonitor should return an `Either` and deal in these type directly?

Reply:
Not sure really, the `progressMonitor` doesn't really _need to_ use an `Either`, whereas the stream does. 

Does @alicefuzier have an opinion?